ambari-server/src/main/java/org/apache/ambari/server/stack/upgrade/Grouping.java

```diff
@@ -249,6 +276,66 @@ private void addTasksToStageInBatches(List<TaskWrapper> tasks, String verb, Upgr
       }
     }
 
+    /**
+     * Utility method to organize and return Rack to Hosts mapping for given rack yaml file
+     *
+     * @param hosts        : All hosts that are part of current group
+     * @param rackYamlFile : file path for yaml containing rack to hosts mapping
+     *                        e.g ambari-server/src/examples/rack_hosts.yaml
+     * @return
+     */
+    private Map<String, Set<String>> organizeHostsByRack(Set<String> hosts, String rackYamlFile) {
```

Comment:
how about adding a unit test for this feature?

Reply:
Done. Thanks for the suggestion @dlysnichenko